.github/workflows/release.yml

```diff
@@ -92,7 +92,7 @@ jobs:
       - name: Compile docs and zip them up
         run: |
           poetry install
-          poetry run -- mkdocs build --no-directory-urls
```

Comment:
sorry about a stupid question because i don't know anything about mkdocs, but what's the effect of just removing this `--no-directory-urls`? because that naively sounds like it would add directory urls back :)

Reply:
Firstly, this snippet generates the offline docs archive where we (still) *don't* want "directory urls". The directory urls are fine on a website, but quite ugly on a file system (it's all files like `revsets/index.html`).

So, the problem this commit solves is with all the *other* invocations of `mkdocs` which are currently compiled as though `--no-directory-urls` is specified. They should be compiled as though `--use-directory-urls` is specified.

Secondly, oh what a naive summer child you are... :innocent: It turns out that removing `--no-directory-urls`, by itself, would be a noop here, since the `offline` plugin forces `--no-directory-urls`. I think it even overrides an explicit `--use-directory-urls` (which is the default anyway). I tried to explain this inside `mkdocs-offline.yml` in this commit. That file has an explicit equivalent of `--no-directory-urls`, which is also actually a no-op, as explained in the file.

The key part of the commit is removing the `offline` plugin for the cases where the docs are generated for the website.

Reply:
Alright, sounds good. Thanks for explaining in such detail. There doesn't seem to be any disadvantage to this PR except for the additional complexity, so I'd say we should merge it.

Reply:
OK, I'll merge it. It's slightly risky, but it's also simplest to test out if links get broken with the live website.